### PR TITLE
Support for Pixtral model (Mistral)

### PR DESCRIPTION
**Pull request by yixin-huang1**
Add support for mistralai/Pixtral ([Feature] Support mistralai/Pixtral #2351).

### CODE REVIEW

**Comment by merrymercy** [comment]
I added you as a co-author in https://github.com/sgl-project/sglang/pull/2500, so your future commits will trigger CI automatically.
Can you fix the lint error? https://sgl-project.github.io/references/contributor_guide.html#format-your-code

**Comment by merrymercy** [comment]
How did you test the correctness of this model locally? Did you compare the logits against HF implements, similar to this one? https://github.com/sgl-project/sglang/pull/2365#discussion_r1872868614

**Comment by merrymercy** [comment]
Please format your code https://sgl-project.github.io/references/contributor_guide.html#format-your-code

File: test/srt/test_pixtral.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                    model_name, torch_dtype=torch.bfloat16, trust_remote_code=True
+                )
+                .eval()
+                .to(self.devices)
+            )
+
+            # Load processor
+            self.processors[key] = AutoProcessor.from_pretrained(model_name)
+
+    async def test_vision_encoder(self):
```
Comment (merrymercy) [inline test/srt/test_pixtral.py]: this one does not need to be async

File: test/srt/test_pixtral.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
+"""
+python3 -m unittest test_pixtral
```
Comment (merrymercy) [inline test/srt/test_pixtral.py:2]: add this to per-commit CI https://github.com/sgl-project/sglang/blob/dc3bee4815183d19761082b9ac90ada72d0151ef/test/srt/run_suite.py#L7

File: test/srt/test_vision_openai_server.py [hunk 3]
```diff
@@ -443,6 +443,79 @@ def setUpClass(cls):
     def test_video_chat_completion(self):
         pass
 
+class TestPixtralServer(TestOpenAIVisionServer):
```
Comment (merrymercy) [inline test/srt/test_vision_openai_server.py]: To save some time on CI, we only need to test one.

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: test/srt/test_pixtral.py (hunk 1 above)
Comment (yixin-huang1) [reply test/srt/test_pixtral.py]: fixed!

File: test/srt/test_pixtral.py (hunk 2 above)
Comment (yixin-huang1) [reply test/srt/test_pixtral.py:2]: added

File: test/srt/test_vision_openai_server.py (hunk 3 above)
Comment (yixin-huang1) [reply test/srt/test_vision_openai_server.py]: yea deleted the other one.

**Comment by merrymercy** [comment]
This PR is wrong. Close this for now. Other contributors please feel free to take this task.
